sklearn/model_selection/tests/test_search.py

```diff
@@ -591,71 +552,209 @@ def test_param_sampler():
         assert_equal([x for x in sampler], [x for x in sampler])
 
 
-def test_randomized_search_grid_scores():
+def check_results_array_types(results, param_keys, score_keys):
+    # Check if the search results' array are of correct types
+    assert_true(all(isinstance(results[param], np.ma.MaskedArray)
+                    for param in param_keys))
+    assert_true(all(results[key].dtype is np.dtype('O')
+                    for key in param_keys))
+    assert_false(any(isinstance(results[key], np.ma.MaskedArray)
+                     for key in score_keys))
+    assert_true(all(results[key].dtype is np.dtype('float64')
+                    for key in score_keys if not key.endswith('rank')))
+
+
+def check_results_keys(results, param_keys, score_keys, n_cand):
+    # Test the search.results_ contains all the required results
+    assert_array_equal(sorted(results.keys()),
+                       sorted(param_keys + score_keys))
+    assert_true(all(results[key].shape[0] == n_cand
```

Comment:
actually, you should check that they are 1d, i.e. that `shape == (n_cand,)` because Pandas won't accept >1d.